staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_discovery.go

```diff
@@ -69,7 +69,7 @@ type DiscoveryAggregationController interface {
 
 	// Spwans a worker which waits for added/updated apiservices and updates
 	// the unified discovery document by contacting the aggregated api services
-	Run(stopCh <-chan struct{})
+	Run(waitCh <-chan struct{}, stopCh <-chan struct{}, discoverySyncedCh chan<- struct{})
```

Comment:
IMO this is kind of an awkward API, to pass in the waitCh just to be waited on at the start line. I think Run should have as its precondition that all the apiservices have been synced and initial HTTP handlers exposed.

Reply:
agree, will move this out